Iris-Pipeline/steps/validate.py

```diff
@@ -0,0 +1,196 @@
+            json_object = json.load(openfile)
+            
+        deploy_model = {
+            "name": model_name,
+        }
+        
+        json_object = json.dumps(deploy_model, indent=4)
+        
+        with open(json_path, "w") as file:
+            file.write(json_object)
+    else:
+        print("Model is not better than current deployed")
+        
+
+def evaluate(
+    model:DecisionTreeClassifier,
+    X_validate: pd.Series,
+    y_validate: pd.Series
+    ) -> None:
+    """
+    Evaluate on the validation set
+    
+    Evaluates the model from the train step against the validation set,
+    to see if the performance is good with unknown data
+    
+    Args:
+        model (DecisionTreeClassifier): trained decision tree classifier,
+            from the last step
+        X_validate (pd.Series): validation features gathered from validation_data.csv
+        y_validate (pd.Series): validation label gathered from validation_data.csv
+    """
+    mlflow.log_metric(
+        "validation_accuracy", metrics.accuracy_score(y_validate, model.predict(X_validate))
+    )
+    mlflow.log_metric(
+        "validation_score", model.score(X_validate, y_validate)
+    )
+    
+    scores = cross_val_score(
+        estimator=model,
+        X=X_validate,
+        y=y_validate,
+        )
+    
+    for count, (score) in enumerate(scores):
+        mlflow.log_metric(
+            f"val_score", score
+        )
+
+
+def save_model(
+    dtc_model: DecisionTreeClassifier,
+    model_name: str,
+    models_path: str
+    ):        
+    mlflow.sklearn.save_model(
+        sk_model=dtc_model,
+        path=models_path,
+        serialization_format=mlflow.sklearn.SERIALIZATION_FORMAT_PICKLE,
+    )
+
+
+@click.command()
+@click.option("--process_run_id")
+@click.option("--train_run_id")
+@click.option("--config_path")
+def task(process_run_id, train_run_id, config_path):
+    """
+    Validate model performance against deployed model
+
+    Orchestrates the validation of model performance. 
+    This is done by first evaluating the model using this function:
+        evaluate(): Evaluates the model using the validation dataset
+    Then it compares the model to the one that is currently deployed using this function:
+        compare_models(): Compares the deployed model metrics to the current model metrics
+    
+    Args:
+        process_run_id (str): run id that the process step generated
+        train_run_id (str): run id that the train step generated
+        config_path (str): path to the configuration file
+    """
+    with mlflow.start_run() as mlrun:
+        with open(config_path, "r", encoding="UTF-8") as ymlfile:
+            cfg = yaml.load(ymlfile, Loader=yaml.FullLoader)
+            
+        json_path = Path("../",cfg["model_path"], cfg["deployment_json_name"])
+        with open(json_path, 'r') as openfile:
+            json_object = json.load(openfile)
+        
+        model_version = train_run_id[:5]    
```

Comment:
For the version, probably the timestamp of when the model is created could be more useful.

Reply:
Elaboration needed. I don’t understand why timestamp could be more useful?

Reply:
It's a choice.
Having a timestamp could help you understand when a model was trained. A random string does not add any particular help.
But, again, it's a choice.

Reply:
I'll look into it